 # The motor housing for the thruster points in the thruster's positive x-direction
# All thrusters are assumed to be pointing in the positive body-frame x-direction
# You MUST define the correct yaw and pitch angle from body-frame TO the thruster frame
properties: {
  mass: 39.825, # [kg]
  # X = F*.8636/W
  # Y = F*.3683/W
  # Z = (X - F*.8636/W)/tan(theta)
  center_of_mass: [0.397, 0.205, -0.17], # Relative to port aft foot [X (m),Y(m),Z(m)]
  dvl: [0.33, 0, 0.44, -90], # Relative to port aft foot [X (m),Y(m),Z(m), yaw (deg)]
  imu: [0.71, 0.19, 0.33], # Relative to port aft foot [X (m),Y(m),Z(m)]
  inertia: [1.32, 2.84, 3.65], # Principle Inertias [kg-m^2]
  depth_fully_submerged: 0.3, # [m]
  thrusters: [ 
  # Pose (from port aft foot): [x (m), y (m), z (m), yaw (deg), pitch (deg)]
  # Type: 0 = CCW spin (symbol: -->), 1 = CW spin (symbol: <--)
    {
      id: 0,
      name: "VPF",
      type: 0,
      enable: true,
      pose: [0.8255, -0.12065, -0.257175, -135.0, 0.0]
    },
    {
      id: 1,
      name: "VSF",
      type: 0,
      enable: true,
      pose: [0.8255, 0.48895, -0.257175, 135.0, 0.0]
    },
    {
      id: 2,
      name: "VPA",
      type: 0, # Should be 1
      enable: true,
      pose: [0.03175, -0.12065, -0.257175, -45.0, 0.0]
    },
    {
      id: 3,
      name: "VSA",
      type: 0, # Should be 1
      enable: true,
      pose: [0.03175, 0.48895, -0.257175, 45.0, 0.0]
    },
    {
      id: 4,
      name: "HPF",
      type: 1,
      enable: true,
      pose: [0.7747, 0, -0.161925, 0.0, 90.0]
    },
    {
      id: 5,
      name: "HSF",
      type: 1, # Should be 0
      enable: true,
      pose: [0.7747, 0.3683, -0.161925, 0.0, 90.0]
    },
    {
      id: 6,
      name: "HPA",
      type: 1, # Should be 0
      enable: true,
      pose: [0.08255, 0, -0.161925, 0.0, 90.0]
    },
    {
      id: 7,
      name: "HSA",
      type: 1,
      enable: true,
      pose: [0.08255, 0.3683, -0.161925, 0.0, 90.0]
    }
  ]
}